packages/eas-cli/src/commands/update/__tests__/republish.test.ts

```diff
@@ -0,0 +1,235 @@
+    jest.mocked(PublishQuery.getCodeSigningInfoFromUpdateGroupAsync).mockResolvedValue({});
+    // Mock mutations to store the new update
+    jest.mocked(ensureBranchExistsAsync).mockResolvedValue({ branchId: updateStub.branch.id });
+    jest.mocked(PublishMutation.publishUpdateGroupAsync).mockResolvedValue([
+      {
+        ...updateStub,
+        id: 'update-new',
+        platform: 'ios',
+      },
+    ]);
+
+    await new UpdateRepublish(flags, commandOptions).run();
+
+    expect(PublishMutation.publishUpdateGroupAsync).toHaveBeenCalledWith(
+      expect.any(Object), // graphql client
+      expect.arrayContaining([
+        expect.objectContaining({
+          branchId: updateStub.branch.id,
+          runtimeVersion: updateStub.runtimeVersion,
+          updateInfoGroup: expect.objectContaining({
+            ios: expect.any(Object),
+          }),
+          gitCommitHash: updateStub.gitCommitHash,
+          awaitingCodeSigningInfo: false,
+        }),
+      ])
+    );
+
+    expect(PublishMutation.setCodeSigningInfoAsync).not.toHaveBeenCalled();
+  });
+
+  it('creates a new update from existing update with codesigning', async () => {
+    const flags = ['--group=1234', '--branch=main', '--message=test-republish'];
+    const codeSigning = {
+      alg: 'alg',
+      keyid: 'keyid',
+      sig: 'sig',
+    };
+
+    mockTestProject();
+    // Mock queries to retrieve the update and code signing info
+    jest.mocked(UpdateQuery.viewUpdateGroupAsync).mockResolvedValue([updateStub]);
+    jest.mocked(PublishQuery.getCodeSigningInfoFromUpdateGroupAsync).mockResolvedValue({
+      ios: codeSigning,
+    });
+    // Mock mutations to store the new update
+    jest.mocked(ensureBranchExistsAsync).mockResolvedValue({ branchId: updateStub.branch.id });
+    jest.mocked(PublishMutation.publishUpdateGroupAsync).mockResolvedValue([
+      {
+        ...updateStub,
+        id: 'update-new',
+        platform: 'ios',
+      },
+    ]);
+    jest.mocked(PublishMutation.setCodeSigningInfoAsync).mockResolvedValue({} as any);
+
+    await new UpdateRepublish(flags, commandOptions).run();
+
+    expect(PublishMutation.publishUpdateGroupAsync).toHaveBeenCalledWith(
+      expect.any(Object), // graphql client
+      expect.arrayContaining([
+        expect.objectContaining({
+          branchId: updateStub.branch.id,
+          runtimeVersion: updateStub.runtimeVersion,
+          updateInfoGroup: expect.objectContaining({
+            ios: expect.any(Object),
+          }),
+          gitCommitHash: updateStub.gitCommitHash,
+          awaitingCodeSigningInfo: true,
+        }),
+      ])
+    );
+
+    expect(PublishMutation.setCodeSigningInfoAsync).toHaveBeenCalledWith(
+      expect.any(Object), // graphql client
+      'update-new',
+      expect.objectContaining(codeSigning)
+    );
+  });
+});
+
+/** Create a new in-memory project, based on src/commands/project/__tests__/init.test.ts */
+function mockTestProject({
```

Comment:
could be worth factoring this out in a follow-up if it's substantially similar to init.test.ts.

Reply:
Yeah, might be worth having this in a test utils section. That way, we don't need most boilerplate codes for each test.